sorted_db/db.go

```diff
@@ -92,17 +101,18 @@ func (db *DB) close() {
 		db.f = nil
 	}
 	db.size = -1
+	return
 }
 
 // Remap DB to the same backing file
 func (db *DB) Remap() error {
-	db.RLock()
+	db.mutex.RLock()
```

Comment:
Is there any reason the `RUnlock` calls here aren't done as part of a `defer`? Would it be a problem to keep the lock until the end of this function?